packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,157 @@
+import userEvent from '@testing-library/user-event';
+
+import { BaseInput } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+import ThemeWrapper from '~src/_helpers/testing/themeWrapper';
+import assertAccessible from '~src/_helpers/testing/assertAccessible.web';
+import { CloseIcon, EyeIcon } from '~components/Icons';
+
+describe('<BaseInput />', () => {
+  it('should render', () => {
+    const { container } = renderWithTheme(<BaseInput label="Enter name" id="name" />);
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render success validation state', () => {
+    const { container } = renderWithTheme(
+      <BaseInput label="Enter name" id="name" validationState="success" successText="Success" />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render error validation state', () => {
+    const { container } = renderWithTheme(
+      <BaseInput label="Enter name" id="name" validationState="error" errorText="Error" />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render with icons', () => {
+    const { container } = renderWithTheme(
+      <BaseInput
+        label="Enter name"
+        placeholder="First Last"
+        id="name"
+        leadingIcon={EyeIcon}
+        trailingIcon={CloseIcon}
+      />,
+    );
+
```

Comment:
shall we check if the icons really rendered or not?

Reply:
I realized we don't pass any a11y labels for these icons

Reply:
Should be fine actually they are mostly for visuals and doesn't have any functionality nor can they take focus.